.github/workflows/check-unlabeled-issues.yml

```diff
@@ -0,0 +1,60 @@
+name: 'Check unlabeled issues'
+on:
+  schedule:
+    # Run every sunday at midnight
+    - cron: '0 0 * * 0'
+
+jobs:
+  check-unlabeled-issues:
+    runs-on: ubuntu-latest
+
+    env:
+      GH_TOKEN: ${{ secrets.GH_TOKEN }}
+
+    steps:
+      - name: Check for unlabeled issues
+        run: |
+          # If any updates are made to the the labels (i.e dropped a label, added a label, etc), just add/remove the label from the lists below.
+          # For instance, if added a "to-do" status label, just add "-label:status:to-do" to the STATUS_LABELS_FILTER.
+          STATUS_LABELS_FILTER='-label:status:requirements -label:status:in-progress -label:status:blocked -label:status:waiting-on-response'
+
+          TYPE_LABELS_FILTER='-label:type:bug -label:type:feature -label:type:docs -label:type:refactor -label:type:help'
+
+          PRIORITY_LABELS_FILTER='label:priority-1-critical -label:priority-2-high -label:priority-3-medium -label:priority-4-low'
+
+          HAS_ISSUES_MISSING_LABELS=false
+
+          ISSUE_BODY="# Label check action\n"
+
+          for FILTER in "$STATUS_LABELS_FILTER" "$TYPE_LABELS_FILTER" "$PRIORITY_LABELS_FILTER"; do
+            # Extract the label type from the filter
+            LABEL_TYPE=$(echo "$FILTER" | cut -d ':' -f 2 | cut -d '-' -f 1)
+
+            # Fetch issues filtered by the label type
+            ISSUES_MISSING_LABEL=$(gh issue list --repo 'renovatebot/renovate' -s open -S "$FILTER" --json number) || { echo "Failed to fetch issues without $LABEL_TYPE labels"; exit 1; }
+
+            if [ "$ISSUES_MISSING_LABEL" != "[]" ]; then
+              HAS_ISSUES_MISSING_LABELS=true
+
+              # Format the output to be a list of issues numbers
+              FORMATTED_OUTPUT=$(echo "$ISSUES_MISSING_LABEL" | jq -r '.[].number' | sed 's/^/- #/')
+          
+              # Count the number of issues to determine if it's singular or plural
+              ISSUE_COUNT=$(echo "$ISSUES_MISSING_LABEL" | jq '. | length')
+              ISSUE_SINGULAR_PLURAL=$(if [ "$ISSUE_COUNT" -eq 1 ]; then echo "issue"; else echo "issues"; fi)
+
+              # Append the list of issues to the new issue body
+              ISSUE_BODY="$ISSUE_BODY## Found $ISSUE_COUNT $ISSUE_SINGULAR_PLURAL missing \`$LABEL_TYPE:\` labels:\n$FORMATTED_OUTPUT\n"
+            fi
+          done
+
+          if [ "$HAS_ISSUES_MISSING_LABELS" ]; then
+            # Create a new issue with the list of issues
+            gh issue create --repo 'renovatebot/renovate' --title "Label check action" --body "$(echo -e "$ISSUE_BODY")" || { echo "Failed to create issue"; exit 1; }
```

Comment:
we should only create one issue and reuse it (reopen, comment).

Reply:
We have another GitHub action that _locks_ closed issues after x amount of days.

Would the "label check" action be able to re-open the issue? I suspect we need to write a special  rule somewhere so the lock action stays away from the "missing label "-issue.

Reply:
It seems that the lock is for issues that have been inactive for more than 30 days, which means that after about the 4th check it will block the issue

From what I've seen in the lock-threads, there is [this condition](https://github.com/dessant/lock-threads#:~:text=exclude%2Dany%2Dissue,defaults%20to%20%27%27) which excludes any issue with a particular label from being locked (maybe something like an `action-label-check` label?), this would also help to find and edit the issue instead of creating a new one. I just don't know if there's a way to prevent other contributors from deliberately using this label to prevent their issues from being locked

Reply:
a normal contributer can't change labels

Reply:
To apply/dismiss labels you need at least triage level access to the repository.

So normal users can not abuse the special label to prevent their issue from getting closed.

Reply:
Oh great, so it's ok to use this approach?

Reply:
I think these are the next steps:

- update lock action so it ignores the label for the "label-issues"-issue
- let the "label issues" action re-open closed issues when needed
- put the Bash code in a separate `.sh` file like @viceice said